tock/employees/tests/test_views.py

```diff
@@ -93,17 +94,19 @@ def test_parse_date(self):
     def test_UserFormViewPermissionForAdmin(self):
         """ Ensure that admin has access to another user's UserFormView
         page """
-        response = self.app.get(
-            reverse('employees:UserFormView', args=["regular.user"]),
-            headers={'X_AUTH_USER': 'aaron.snow@gsa.gov'},
+        self.login(username="aaron.snow", is_superuser=True)
+        response = self.client.get(
+            reverse('employees:UserFormView', args=["regular.user"])
         )
         # Check that inital data for UserDate Populates
-        self.assertEqual(
-            response.html.find('input', {'id': 'id_start_date'})['value'],
-            '2014-01-01')
-        self.assertEqual(
-            response.html.find('input', {'id': 'id_end_date'})['value'],
-            '2016-01-01')
+        self.assertContains(
+            response,
+            '<input class="datepicker" id="id_start_date" name="start_date" type="text" value="2014-01-01" />'
+        )
+        self.assertContains(
+            response,
+            '<input class="datepicker" id="id_end_date" name="end_date" type="text" value="2016-01-01" />'
+        )
```

Comment:
In looking at these I think this is okay because as you said, this is going off of the template response now instead of the HTTP response directly.

Reply:
By the way, it might be helpful to pass `html=True` to [`assertContains()`](https://docs.djangoproject.com/en/2.0/topics/testing/tools/#django.test.SimpleTestCase.assertContains) to make the tests less fragile:

> Set `html` to `True` to handle text as HTML. The comparison with the response content will be based on HTML semantics instead of character-by-character equality. Whitespace is ignored in most cases, attribute ordering is not significant.